### PR TITLE
Allow single condition in `ensures` clause

### PR DESCRIPTION
**Pull request by ranjitjhala**
e.g. to write 

```rust
#[flux::sig(fn(b:bool) ensures b)]
pub fn assume(b: bool) {
    if !b {
        panic!("assume fails")
    }
}
```

### CODE REVIEW

File: crates/flux-syntax/src/surface.rs [hunk 1]
```diff
@@ -153,14 +153,22 @@ pub struct FnSig<R = ()> {
     pub args: Vec<Arg<R>>,
     /// example `i32{v:v >= 0}`
     pub returns: FnRetTy<R>,
-    /// example: `*x: i32{v. v = n+1}`
-    pub ensures: Vec<(Ident, Ty<R>)>,
+    /// example: `*x: i32{v. v = n+1}` or just `x > 10`
+    pub ensures: Ensures<R>,
```
Comment (nilehmann) [inline crates/flux-syntax/src/surface.rs:157]: Any reason to not make this a list of constraints (where a constraint is either `x: T` or `pred`) like in the lower levels?

File: crates/flux-tests/tests/neg/surface/assume00.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+
+#[flux::sig(fn(b:bool[true]))]
+pub fn assert(_b: bool) {}
```
Comment (nilehmann) [inline crates/flux-tests/tests/neg/surface/assume00.rs:5]:
```suggestion
#[flux::sig(fn(bool[true]))]
pub fn assert(_: bool) {}
```

File: crates/flux-tests/tests/neg/surface/assume00.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+#[flux::sig(fn(b:bool) ensures b)]
+pub fn assume(b: bool) {
+    if !b {
+        panic!("assume fails")
+    }
+}
```
Comment (nilehmann) [inline crates/flux-tests/tests/neg/surface/assume00.rs:12]: We should add a test where the `ensures` fails

```rust
#[flux::sig(fn(x: i32) ensures x > 0)]
pub fn fail_ensures(x: i32) { }
```

or maybe

```rust
#[flux::sig(fn(b: bool) ensures b)]
pub fn bad_assume(b: bool) { }
```

**Comment by nilehmann** [comment]
This also makes me think about `panic` in general. Normally you would like to prove the absence of panics, but the soroban example may be a compelling use case where you want the opposite. In general, you probably want both

```rust
fn assume(b: bool) ensures b {
    if !b { panic!("failed assume") }
}

fn unreachable<T>() -> T requires false {   // or `fn unreachable() -> ! requires false` in nightly
    panic!("entered unreachable code")
}
```

File: crates/flux-syntax/src/surface.rs (hunk 1 above)
Comment (ranjitjhala) [reply crates/flux-syntax/src/surface.rs:157]: duh. let me do that!

File: crates/flux-tests/tests/neg/surface/assume00.rs (hunk 3 above)
Comment (ranjitjhala) [reply crates/flux-tests/tests/neg/surface/assume00.rs:12]: good call!

**Comment by ranjitjhala** [comment]
ok I think I've fixed the above.

**Review by nilehmann — APPROVED** [review]
:shipit:

**Pull request merged**